.github/workflows/ci.yml

```diff
@@ -12,6 +12,15 @@ on:
     types: [published, edited]
 
 jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.8"
+      - uses: pre-commit/action@v3.0.1
```

Comment:
@andhus my issue here is that this action is in maintenance mode in favor of their own CI service (which is free for open source so i suggest switching to it asap)
and it uses their pre-commit config instead of the one inside of dirhash repo (so it will never find the issues you will see when running pre-commit locally)

Reply:
> my issue here is that this action is in maintenance mode in favor of their own CI service (which is free for open source so i suggest switching to it asap)

Well, it is a [super light-weight action](https://github.com/pre-commit/action/blob/main/action.yml) that we could just copy-past it in here if it was removed. It is also very fast with caching etc. Even though I really appreciate asottile's work with pre-commit, I don't see the need to involve yet another thrid-party service, when github workflows solve it 100% with a few lines. (Even if it is free.)

> and it uses their pre-commit config instead of the one inside of dirhash repo (so it will never find the issues you will see when running pre-commit locally)

No, that is incorrect. It just runs pre-commit, so your config. Check logs to verify: https://github.com/andhus/dirhash-python/actions/runs/8713046761/job/23900662905#step:4:112